### PR TITLE
Paginated list selector changes for editing linked products for a grouped product

### PR DESCRIPTION
**Pull request by jaclync**
Prerequisite for editing linked products for a grouped product in #2199 

## Changes

- Removed `PaginatedListSelectorViewController.xib` and added a table view manually in `PaginatedListSelectorViewController` so that the table view style (`.grouped`/`.plain`) can be configured in `PaginatedListSelectorViewProperties`. A table view's `tableViewStyle` cannot be changed after initialization from the xib. In #2199, I'm planning to use `PaginatedListSelectorViewController` as a child view controller with `.plain` table view style for the "Add Products" screen that is similar to the Products tab
- Added `separatorStyle: UITableViewCell.SeparatorStyle` to `PaginatedListSelectorViewProperties` since product cell used in #2199 does not use the built-in cell separator
- Set the table view style to `.grouped` for existing usage: editing product shipping classes and tax classes
- In `PaginatedListSelectorViewController`, updated UI setup based on the table view properties in `PaginatedListSelectorViewProperties`
- Since we are displaying a row type that doesn't have a xib (`ProductsTabProductTableViewCell`) in #2199, added a check to register the cell with class if the nib is not found

## Testing

A sanity check on the affected screens would be great!

- Go to the Products tab
- Tap on a simple physical product
- Tap on the price row, and tap on the tax class row --> the paginated tax classes should be shown as in `develop`, and selection should work as expected
- Go back to product details
- Tap on the shipping row, and tap on the shipping class row --> the paginated shipping classes should be shown as in `develop`, and selection should work as expected

## Example screenshots

No user-facing changes are expected!

\ | before | after
-- | -- | --
editing tax classes | Simulator Screen Shot - iPhone 11 - 2020-07-02 at 10 53 46 | Simulator Screen Shot - iPhone 11 - 2020-07-02 at 10 46 37
editing shipping classes | Simulator Screen Shot - iPhone 11 - 2020-07-02 at 10 53 56 | Simulator Screen Shot - iPhone 11 - 2020-07-02 at 10 46 47

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jaclync** [comment]
@woocommerce/ios-developers this is ready for review, with another PR https://github.com/woocommerce/woocommerce-ios/pull/2505 depending on this one!

File: WooCommerce/Classes/ViewRelated/ListSelector/PaginatedListSelectorViewController.swift [hunk 1]
```diff
@@ -235,10 +235,26 @@ private extension PaginatedListSelectorViewController {
         tableView.rowHeight = UITableView.automaticDimension
         tableView.backgroundColor = .listBackground
 
+        tableView.separatorStyle = viewProperties.separatorStyle
+
+        if viewProperties.tableViewStyle == .plain {
+            // Removes extra header spacing in ghost content view.
+            tableView.estimatedSectionHeaderHeight = 0
+            tableView.sectionHeaderHeight = 0
+        }
```
Comment (pmusolino) [inline WooCommerce/Classes/ViewRelated/ListSelector/PaginatedListSelectorViewController.swift:244]: Seems that this condition need to be executed always, also when the table view style is grouped, or this is what happens if compared to develop:

| Develop - ghost tableview            |  This branch - ghost tableview |
:-------------------------:|:-------------------------:
Simulator Screen Shot - iPhone 11 Pro - 2020-07-14 at 14 33 40 | Simulator Screen Shot - iPhone 11 Pro - 2020-07-14 at 14 27 31

This happens both in Tax classes and Shipping classes.

File: WooCommerce/Classes/ViewRelated/ListSelector/PaginatedListSelectorViewController.swift [hunk 2]
```diff
@@ -235,10 +235,26 @@ private extension PaginatedListSelectorViewController {
...
+
+        view.addSubview(tableView)
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        view.pinSubviewToSafeArea(tableView)
+
         registerTableViewCells()
     }
 
     func registerTableViewCells() {
+        guard Bundle.main.path(forResource: rowType.classNameWithoutNamespaces, ofType: "nib") != nil else {
```
Comment (pmusolino) [inline WooCommerce/Classes/ViewRelated/ListSelector/PaginatedListSelectorViewController.swift:254]: 👍

File: WooCommerce/Classes/ViewRelated/ListSelector/PaginatedListSelectorViewController.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/ListSelector/PaginatedListSelectorViewController.swift]: Thanks @pmusolino for noticing this issue! yes it does look like this is needed for grouped table view style as well. updated in https://github.com/woocommerce/woocommerce-ios/pull/2492/commits/a9485bdf4a222c40671b2eadf3e720662b23e071 and screenshots on price tax classes:

`develop` | PR branch
-- | --
Simulator Screen Shot - iPhone 11 - 2020-07-15 at 09 47 08 | Simulator Screen Shot - iPhone 11 - 2020-07-15 at 09 49 51

**Comment by jaclync** [comment]
thank you for finding the ghost view issue @pmusolino ! I already loaded the tax classes on my site in the app so I missed the loading screen. I updated the PR for another look 😄

**Pull request merged**